code/harness/Tester.cpp

```diff
@@ -194,6 +197,53 @@ int Tester::umount_device() {
   return SUCCESS;
 }
 
+int Tester::mount_snapshot() {
+  if (mount(SNAPSHOT_PATH, MNT_MNT_POINT, fs_type.c_str(), 0, NULL) < 0) {
+    return MNT_MNT_ERR;
+  }
+  return SUCCESS;
+}
+
+int Tester::umount_snapshot() {
+  if (umount(MNT_MNT_POINT) < 0) {
+    return MNT_UMNT_ERR;
+  }
+  return SUCCESS;
+}
+
+int Tester::mapCheckpointToSnapshot(int checkpoint) {
+  if (checkpointToSnapshot_.find(checkpoint) != checkpointToSnapshot_.end()) {
+    return -1;
+  }
+  std::string snapshot(SNAPSHOT_PATH);
+  checkpointToSnapshot_[checkpoint] = (char *) malloc(sizeof(char)*30);
+  strcpy(checkpointToSnapshot_[checkpoint], snapshot.c_str());
+  std::cout << "Mapping " << SNAPSHOT_PATH << " to checkpoint " << checkpoint << std::endl;
+  return 0;
+}
+
+int Tester::getNewDiskClone(int checkpoint) {
+  string new_snapshot_path;
+  string path(SNAPSHOT_PATH);
+  string device_number = path.substr(path.rfind('_'));
+  string snapshot_number = to_string(checkpoint+2);
+  new_snapshot_path = "/dev/cow_ram_snapshot";
+  new_snapshot_path += snapshot_number;
+  new_snapshot_path += device_number;
+  // Finally set SNAPSHOT_PATH to the new snapshot path
+  strcpy(SNAPSHOT_PATH, new_snapshot_path.c_str());
+  // TODO (P.S.) Make it a fs specific command and get it here
+  string command = fs_specific_ops_->GetNewUUIDCommand(new_snapshot_path);
+  // string command = "yes | btrfstune -u ";
+  // command += new_snapshot_path;
+  system(command.c_str());
+  return 0;
+}
+
+void Tester::getCompleteRunDiskClone() {
+  SNAPSHOT_PATH = checkpointToSnapshot_[0];
```

Comment:
Could you add a comment at the declaration of checkpointToSnapshot_ in `Tester.h` that the 0th element is the default for testing?